### ISSUE TITLE
X windows from other workspaces showing on when a new window is mapped

### ISSUE THREAD

**Issue by pjones** [ch11ng/exwm#446]
I've been using EXWM for about a month but just noticed this because I started using EXWM workspaces instead of using elscreen with a single frame.

Here is what I'm seeing:

  1. Switch to a workspace and open an X window (browser, openscad, etc.)

  2. Switch to a different workspace and open another X window

  3. All other X windows are drawn behind the one opened in step 2.  The window configuration hasn't changed, it's just that all windows are now obscured by X windows from other workspaces.

 To fix this I have to manually visit each workspace showing an X window.  As I switch away from a workspace its X windows stay with it, clearing it from the screen.

I'm using EXWM from Git two commits ago.  Emacs 25.3.  This happens with and without a compositor running.  Here is my [EXWM configuration](https://github.com/pjones/emacsrc/blob/master/modes/exwm-conf.el).

Thanks.

**Comment by pjones** [comment]
Oh, I just realized that this only happens when I open a floating X window in step 2.  Maybe I have some sort of floating configuration set wrong?

**Comment by pjones** [comment]
Setting `exwm-manage-force-tiling` to `t` prevents this from happening.  When I get some time over the next couple of weeks I'll debug the floating functions and see where this is coming from.

**Comment by pjones** [comment]
I tracked this down.  

I have a function in `after-make-frame-functions` that makes sure things like the menu bar are turned off by calling functions like `(menu-bar-mode -1)`.  This would immediately cause all X windows on the current monitor to be drawn on top of the current frame.

I made the function smarter by only turning those modes off if they were actually on.  This seems to have fixed my issue.
